src/common/FrameManager.ts

```diff
@@ -347,6 +392,9 @@ export class FrameManager extends EventEmitter {
 
   _onFrameDetached(frameId: string): void {
     const frame = this._frames.get(frameId);
+    if (frame.isOOPFrame()) {
```

Comment:
why is it needed?

Reply:
If the frame becomes OOP, it will be detached from the parent targets session but it's still the same frame. Therefore it shouldn't actually be removed.

Reply:
I'll add a comment to explain what's going on.

Reply:
And what happens if a isOOPFrame==true frame is detached for real?

Reply:
I updated the change to address this. Thanks!